eventuals/expected.h

```diff
@@ -96,10 +99,22 @@ class expected : public tl::expected<Value_, Error_> {
               std::runtime_error>>,
       Errors>;
 
-  template <typename Arg, typename K>
+  template <typename Arg, typename Errors, typename K>
   auto k(K k) && {
     return ExpectedToEventual(std::move(*this))
-        .template k<Value_>(std::move(k));
+        .template k<
+            Value_,
+            tuple_types_union_t<
+                std::tuple<
+                    std::conditional_t<
+                        std::disjunction_v<
+                            CheckErrorsTypesForVariant<std::decay_t<Error_>>,
```

Comment:
I don't want to propagate `std::variant` for error types right now, so this basically needs to be a union of `Errors` and the error type of the `expected`, i.e., `Error_`. In the case we have `std::variant<E1, E2>` as the error type of the `expected` then we should extract each of those into a tuple and then union that with `Errors`. Make sense?